warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionEvaluator.java

```diff
@@ -190,8 +219,16 @@ public Collection<String> evaluate() {
 
                 // Iterate over each collection of offsets (grouped by field) and try to find one that satisfies the phrase/adjacency
                 for (String field : offsetsByField.keySet()) {
-                    if (!fields.isEmpty() && !fields.contains(field)) {
-                        continue;
+                    if (!fields.isEmpty()) {
+                        // account for trailing dot notation
+                        String compareField = field;
+                        int dotIndex = field.indexOf(".");
+                        if (dotIndex > -1) {
+                            compareField = field.substring(0, dotIndex);
+                        }
```

Comment:
This feels ripe for a static utility method, similar to the ASTHelper's deconstruct identifier method.